qiskit/transpiler/passes/scheduling/alap.py

```diff
@@ -138,7 +138,8 @@ def run(self, dag):
             for bit in node.qargs:
                 delta = t0 - idle_before[bit]
                 if delta > 0:
-                    new_dag.apply_operation_front(Delay(delta, time_unit), [bit], [])
+                    if self.target is None or (bit_indices[bit],) in self.target.get("delay", []):
```

Comment:
Do we have a better way to check if an instruction is supported in a target? @mtreinish

Reply:
I think you can use this method.
https://github.com/Qiskit/qiskit-terra/blob/908adb34323682cf48a8bbc711d01f651dd26932/qiskit/transpiler/target.py#L568-L573

Reply:
Umm, it may not work for a target that does not support only the delay instruction.

Reply:
There is the `instruction_supported` method you can use to do this lookup: https://qiskit.org/documentation/stubs/qiskit.transpiler.Target.instruction_supported.html#qiskit.transpiler.Target.instruction_supported it should handle all the edge cases around doing the lookup correctly